Basic-Car-Maintenance/Shared/Onboarding/Views/WelcomeViewAddVehicle.swift

```diff
@@ -7,105 +7,218 @@
+        LabeledContent {
+            TextField("Vehicle \(title)", text: text)
+                .frame(width: 200)
+        } label: {
+            HStack {
+                Text(title)
+                Spacer()
+            }
+        }
+        .showClearButton(text)
+    }
+    
+    private var bottomText: some View {
+        Text("You can edit more data about the vehicle in the 'Settings' tab.")
+            .foregroundStyle(.gray)
+            .padding(.horizontal, 25)
+    }
+    
+    private var addVehicleButton: some View {
+        Button {
+            addVehicle()
+        } label: {
+            Text("Add Vehicle")
+                .fontWeight(.bold)
+                .foregroundStyle(.white)
+                .frame(maxWidth: .infinity)
+                .padding(.vertical, 14)
+                .background(Color("basicGreen").gradient, in: .rect(cornerRadius: 12))
+                .contentShape(.rect)
+        }
+        .padding(15)
+        .padding(.horizontal, 15)
+    }
+    
+    private func addVehicle() {
+        if vehicleName.isEmpty {
+            alertType = .emptyName
+            showAlert = true
+        } else if vehicleMake.isEmpty {
+            alertType = .emptyMake
+            showAlert = true
+        } else if vehicleModel.isEmpty {
+            alertType = .emptyModel
+            showAlert = true
+        } else {
             
-            Spacer(minLength: 10)
+            let newVehicle = Vehicle(name: vehicleName, make: vehicleMake, model: vehicleModel)
             
-            Button {
-//                isFirstTime = false
-            } label: {
-                Text("Welcome 🥳")
-                    .fontWeight(.bold)
-                    .foregroundStyle(.white)
-                    .frame(maxWidth: .infinity)
-                    .padding(.vertical, 14)
-                    .background(Color("basicGreen").gradient, in: .rect(cornerRadius: 12))
-                    .contentShape(.rect)
+            Task {
+                do {
+                    try await viewModel.addVehicle(newVehicle)
+                    print("Vehicle \(newVehicle.name) added to firebase successfully!")
+                    alertType = .vehicleAdded(name: vehicleName)
+                    showAlert = true
+                } catch {
+                    alertType = .error(message: "Failed to add vehicle. Please try again.")
+                    showAlert = true
+                }
             }
-            .padding(15)
-            .padding(.horizontal, 15)
         }
-        .frame(maxWidth: .infinity, maxHeight: .infinity)
-        .background {
-            Color(UIColor.secondarySystemBackground)
-                .ignoresSafeArea()
-        }
-        .navigationBarBackButtonHidden(true)
-        .toolbar {
-            ToolbarItem(placement: .topBarLeading) {
-                Button {
-                    dismiss()
-                } label: {
+    }
+}
+
+struct TextFieldClearButton: ViewModifier {
```

Comment:
this should be in it's own file, because it can be used other places than only here
I fixed this